### PR TITLE
feat: Add support for sourceTypeOverride and sourceLocationOverride

### PR DESCRIPTION
**Pull request by yamao-latte**
This commit introduces two new optional parameters:

sourceTypeOverride: Allows overriding the source input type defined in the build project for the current build. Valid values include NO_SOURCE, CODECOMMIT, CODEPIPELINE, GITHUB, S3, BITBUCKET, and GITHUB_ENTERPRISE.
sourceLocationOverride: Enables specifying a location that overrides the source location defined in the build project for the current build.

These parameters provide flexibility in customizing the source settings for individual builds, without modifying the underlying build project configuration.

Issue #164 

*Description of changes:*
- Added `sourceTypeOverride` parameter to allow users to override the source input type defined in the build project.
  - Updated the `action.yml` file to include the `sourceTypeOverride` parameter with a description and set it as optional.
  - Modified the relevant code sections to handle the `sourceTypeOverride` parameter and pass its value to the CodeBuild API when starting a build.
- Introduced `sourceLocationOverride` parameter to enable users to specify a custom source location for the current build.
  - Updated the `action.yml` file to include the `sourceLocationOverride` parameter with a description and set it as optional.
  - Modified the relevant code sections to handle the `sourceLocationOverride` parameter and pass its value to the CodeBuild API when starting a build.
- Updated the README.md file to document the new `sourceTypeOverride` and `sourceLocationOverride` parameters, providing examples and explaining their usage.

These changes enhance the flexibility and usability of the Action, allowing users to customize their build source settings on a per-build basis and enabling integration with various source providers, including GitHub Enterprise.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Comment by natanaelfiorilla-scopely** [comment]
Please, review this PR.

**Review by kadeboer — APPROVED** [review]
I could also really use this, and it looks sound to me.

**Comment by avx-rodmans** [comment]
I need this as well

**Comment by taoyong-ty** [comment]
Thanks for making this change! taking a look now

File: code-build.js [hunk 1]
```diff
@@ -205,6 +205,13 @@ function githubInputs() {
       : process.env[`GITHUB_SHA`]);
 
   assert(sourceVersion, "No source version could be evaluated.");
+
+  const sourceTypeOverride = 
+    core.getInput("source-type-override", { required: false, }) || undefined;
+
```
Comment (taoyong-ty) [inline code-build.js:211]: These two new attributes need to be added to the returned struct [here](https://github.com/aws-actions/aws-codebuild-run-build/blob/main/code-build.js#L258-L275)

File: code-build.js [hunk 2]
```diff
@@ -296,8 +305,8 @@ function inputs2Parameters(inputs) {
   const sourceOverride = !disableSourceOverride
     ? {
         sourceVersion: sourceVersion,
-        sourceTypeOverride: "GITHUB",
-        sourceLocationOverride: `https://github.com/${owner}/${repo}.git`,
+        sourceTypeOverride: sourceTypeOverride || "GITHUB",
+        sourceLocationOverride: sourceLocationOverride || `https://github.com/${owner}/${repo}.git`,
```
Comment (taoyong-ty) [inline code-build.js:309]: could you please add an unit test ? 

you can find an example here: https://github.com/aws-actions/aws-codebuild-run-build/blob/main/test/code-build-test.js#L410-L421

**Review by taoyong-ty — CHANGES_REQUESTED** [review]

**Review by taoyong-ty — APPROVED** [review]

**Review by taoyong-ty — APPROVED** [review]

**Pull request merged**